app/src/main/java/com/nervousfish/nervousfish/modules/database/GsonKeyAdapter.java

```diff
@@ -18,50 +18,43 @@
 final class GsonKeyAdapter extends TypeAdapter<IKey> {
 
     /**
-     * Constructor for the {@code GsonKeyAdapter} class.
-     */
-    GsonKeyAdapter() {
-        super();
-    }
-
-    /**
      * {@inheritDoc}
      */
     @Override
-    public void write(final JsonWriter writer, final IKey key) throws IOException {
-        writer.beginArray();
+    public void write(final JsonWriter jsonWriter, final IKey t) throws IOException {
```

Comment:
Because the parameter name of the interface is called t, see the javadoc, and it can be confusing if the parameter name of the superclass (the interface) is different from the parameter name of the implementation